pkg/porter/cnab.go

```diff
@@ -234,11 +234,20 @@ func (o *sharedOptions) parseParams() error {
 
 // parseParamSets parses the variable assignments in ParameterSets.
 func (o *sharedOptions) parseParamSets(p *Porter) error {
-	parsed, err := p.loadParameterSets(o.ParameterSets)
-	if err != nil {
-		return errors.Wrapf(err, "unable to process provided parameter sets: %v", o.ParameterSets)
+	if len(o.ParameterSets) > 0 {
+		// Load the manifest as it may be needed to determine if any
+		// parameters are of the Porter-exclusive type of 'file'
+		err := p.LoadManifestFrom(o.File)
```

Comment:
This results in the manifest being loaded twice because we are loading it here (sometimes) and also in `p.applyDefaultOptions` (always).  I think this inconsistency may lead to bugs where we aren't sure when we can rely on the manifest being set, and by when.

Reply:
Good call.  My strategy on consolidation is to move the `p.applyDefaultOptions` into the `Validate` method on `sharedOptions`, to be sure the manifest is loaded prior to parsing parameters.  This also has the advantage of removing duplicated invocations from the action files (install.go, etc.)  WDYT?